docs/diagrams/ModelClassDiagram.puml

```diff
@@ -3,48 +3,53 @@
 skinparam arrowThickness 1.1
 skinparam arrowColor MODEL_COLOR
 skinparam classBackgroundColor MODEL_COLOR
+skinparam classAttributeIconSize 0
 
 Package Model <<Rectangle>>{
-Class "<<interface>>\nReadOnlyAddressBook" as ReadOnlyAddressBook
+Class "<<interface>>\nReadOnlyResidentBook" as ReadOnlyResidentBook
 Class "<<interface>>\nReadOnlyUserPrefs" as ReadOnlyUserPrefs
 Class "<<interface>>\nModel" as Model
-Class AddressBook
-Class ModelManager
+Class ResidentBook
+Class ModelManager {
+- visibleFields: ObservableList<String>
+- hiddenFields: ObservableList<String>
```

Comment:
I'm still not sure if we should add this to the UML diagram. I'm not sure if my analogy is correct but if we add these, should we add the implementation of using `FilteredList`?

In the AB3 DG, they did not include `FilteredList`, but they made references as below:

> stores the currently ‘selected’ Person objects (e.g., results of a search query) as a separate filtered list which is exposed to outsiders as an unmodifiable ObservableList<Person> that can be ‘observed’ e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.